CK3toEU4/Source/EU4World/FlagFoundry/Warehouse.cpp

```diff
@@ -0,0 +1,186 @@
+
+		Log(LogLevel::Debug) << "warehouse pattern enter on " << newImageBlock.patternName;
+
+		// Do we have an emblem like this stored?
+		auto matchedEmblem = false;
+		if (directoryName == "colored_emblems")
+			for (const auto& storedImage: coloredStorage)
+			{
+				if (newImageBlock == storedImage)
+				{
+					Log(LogLevel::Debug) << "warehouse colored texture match on cache";
+					toReturn.emplace_back(std::pair(emblem, storedImage.imageData)); // We have a match of patternName and all possible colors.
+					matchedEmblem = true;
+					break;
+				}
+			}
+		else if (directoryName == "textured_emblems")
+			for (const auto& storedImage: texturedStorage)
+			{
+				if (newImageBlock == storedImage)
+				{
+					Log(LogLevel::Debug) << "warehouse textured texture match on cache";
+					toReturn.emplace_back(std::pair(emblem, storedImage.imageData)); // We have a match of patternName and all possible colors.
+					matchedEmblem = true;
+					break;
+				}
+			}
+		if (matchedEmblem)
+			continue; // We're done here.
+
+		// Do we have a base texture in storage from which we can craft an image?
+		if (!baseTextures.count(newImageBlock.patternName))
+		{
+			// Load the texture from disk.
+			if (!commonItems::DoesFileExist(imageFolder + directoryName + "/" + newImageBlock.patternName))
+			{
+				Log(LogLevel::Warning) << "Emblem uses invalid texture: " << newImageBlock.patternName << ", defaulting to black.";
+				toReturn.emplace_back(std::pair(emblem, Magick::Image(Magick::Geometry(128, 128), Magick::ColorRGB(0, 0, 0))));
+				continue;
+			}
+			try
+			{
+				Log(LogLevel::Debug) << "warehouse texture loading from disk: " << imageFolder + directoryName + "/" + newImageBlock.patternName;
+				baseTextures.insert(std::pair(newImageBlock.patternName, Magick::Image(imageFolder + directoryName + "/" + newImageBlock.patternName)));
+			}
+			catch (std::exception& e)
+			{
+				Log(LogLevel::Warning) << "Emblem uses invalid texture: " << newImageBlock.patternName << ", defaulting to black. (" << e.what() << ")";
+				toReturn.emplace_back(std::pair(emblem, Magick::Image(Magick::Geometry(128, 128), Magick::ColorRGB(0, 0, 0))));
+				continue;
+			}
+		}
+
+		newImageBlock.imageData = baseTextures[newImageBlock.patternName];
+		Log(LogLevel::Debug) << "texture is sized: " << newImageBlock.imageData.size().width() << "x" << newImageBlock.imageData.size().height();
+
+		// Send to coloring department.
+		Log(LogLevel::Debug) << "warehouse texture sending to recolorer";
+		newImageBlock.imageData = recolorer.craftTextureImage(newImageBlock);
+
+		// Store for posterity.
+		texturedStorage.emplace_back(newImageBlock);
+
+		// And fulfill order
+		toReturn.emplace_back(std::pair(emblem, newImageBlock.imageData));
+	}
+	Log(LogLevel::Debug) << "warehouse returning textures";
+	return toReturn;
+}
+
+bool EU4::coloredImage::operator==(const coloredImage& rhs)
+{
+	if (this->patternName != rhs.patternName)
```

Comment:
Is there a reason you're using this to access the members instead of doing so directly?